src/hooks/useSuggestions.ts

```diff
@@ -1,6 +1,9 @@
 import { useEffect, useState } from 'react'
 
 import { SickObj, suggestionAPI } from '../apis/suggestion'
+import { CacheRepository } from '../store/CacheRepository'
+
+const cacheRepository = new CacheRepository<SickObj>()
```

Comment:
suggestionAPI는 같은 cacheRepository를 공유하고, 만약 다른 api가 있다면 api마다 각각 cacheRepository 인스턴스를 생성해 공유한다고 이해하면 될까요?

Reply:
해당 api를 캐싱하고 싶은 곳에서 각각 인스턴스를 생성해서 사용하는 방식입니다!

```js
// src/hooks/useSuggestions.ts

suggestionCacheRepository = new CacheRepository<SickObj>()
```

```js
// src/hooks/example.ts

exampleCacheRepository = new CacheRepository<ExampleData>()
```


export는 하지 않아서 useSuggestions 훅 내에서만 사용하는 걸 알 수 있지만, 다시 생각해보니 다음과 같은 네이밍도 괜찮을 것 같네요.

```suggestion
const suggestionCacheRepository = new CacheRepository<SickObj>()
```

Reply:
답변 감사합니다!👍👍